clang/lib/AST/StmtPrinter.cpp

```diff
@@ -497,6 +497,19 @@ void StmtPrinter::VisitReturnStmt(ReturnStmt *Node) {
   if (Policy.IncludeNewlines) OS << NL;
 }
 
+static void PrintGCCAsmString(raw_ostream &OS, Expr*E) {
+  if(E->getDependence()) {
+    OS << "<<dependent expr>>";
+    return;
+  }
+
+  std::string Str = GCCAsmStmt::ExtractStringFromGCCAsmStmtComponent(E);
```

Comment:
Do we really want to do this instead of printing the expression?  Seems weird that `-ast-print` would be changing the value here.